### PR TITLE
Bugfix: Finding git directories.

### PR DESCRIPTION
**Pull request by gabyx**
To review. Not yet perfect.

```
GIT_DIR=$(cd "$EXISTING" && GIT_DISCOVERY_ACROSS_FILESYSTEM=0 git rev-parse --git-dir)
```
needs revisit. Return relative path??

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -4101,6 +4101,47 @@ setup_automatic_update_checks() {
...
+    for EXISTING in $REPOSITORY_LIST; do
+        unset IFS
+
+        if [ -f "$EXISTING" ]; then
+            # Strip HEAD file
+            EXISTING=$(dirname "$EXISTING")
+        fi
+
+        # Go to the root git dir (works in bare and non-bare repositories)
+        GIT_DIR=$(cd "$EXISTING" && GIT_DISCOVERY_ACROSS_FILESYSTEM=0 git rev-parse --git-dir 2>/dev/null)
```
Comment (rycus86) [inline install.sh]: Is the `GIT_DISCOVERY_ACROSS_FILESYSTEM=0` necessary here?

File: install.sh [hunk 2]
```diff
@@ -4101,6 +4101,47 @@ setup_automatic_update_checks() {
...
+        if [ -n "$GIT_DIR" ] && ! echo "$EXISTING_REPOSITORY_LIST" | grep -q "$GIT_DIR"; then
+            EXISTING_REPOSITORY_LIST=$(printf '%b' "$GIT_DIR\n$EXISTING_REPOSITORY_LIST")
```
Comment (rycus86) [inline install.sh]: This should probably be `printf 'pattern' "$VAR1" "$VAR2"`

**Comment by rycus86** [comment]
Do we have any other is_git_dir checks anywhere? Might be good to keep them consistent and doing the same thing.

**Comment by rycus86** [comment]
Thanks a lot for looking into this btw!

File: install.sh (hunk 1 above)
Comment (gabyx) [reply install.sh]: I would put it in, because someone accidentally set that, I just disable it.

File: install.sh (hunk 2 above)
Comment (gabyx) [reply install.sh]: printf not so portable `printf "%b" "a\nb"`, I changed to `echo` with newline in the script

**Comment by gabyx** [comment]
Please see the changes now.

Btw: Would it make sense to collaborate all copy&paste functions into a `general.sh` which we also inject during commiting. so we only need to fix bugs once, would you mind refactoring this? Would help in the future.

**Comment by rycus86** [comment]
Changes look good, just waiting for Travis then I'll merge it.

I'll think about the shared snippet to insert, it's not only a good way to share code but also to share bloat that some scripts wouldn't have needed otherwise. 🙂

**Comment by gabyx** [comment]
>Do we have any other is_git_dir checks anywhere? Might be good to keep them consistent and doing the same thing.

we only have `is_git_repo` tests and these are ok I think.
Other git_dir checks I dont know

**Comment by gabyx** [comment]
I have not tested it by me :-) (on windows, unfortunately.)
If it does not work, I check tonight...

**Pull request merged**